pilot/pkg/model/push_context.go

```diff
@@ -182,6 +189,17 @@ func (first *UpdateRequest) Merge(other *UpdateRequest) UpdateRequest {
 	return merged
 }
 
+// PushRequest defines a request to push to proxies
+// It is used after doPush and before pushQueue
+type PushRequest struct {
```

Comment:
why do we need another structure? I like the name of this structure. Can we merge update request and push request?

Reply:
It is difficult, as in different procedure, they need diff configs.

But i will have a look to make it more elegant.